codegen/core/src/main/scala/com/lightbend/akkasls/codegen/DescriptorSet.scala

```diff
@@ -83,4 +78,28 @@ object DescriptorSet {
   private val descriptorslogger = Logger.getLogger(classOf[Descriptors].getName)
   descriptorslogger.setLevel(Level.OFF); // Silence protobuf
 
+  /**
+   * This method accumulates `FileDescriptor`s to provide
+   * all the necessary dependencies for each call to FileDescriptor.buildFrom.
+   * Otherwise placeholders (mocked references) get created instead and
+   * these can't function as proper dependencies. Chiefly as imports.
+   *
+   * see allowUnknownDependencies  per https://github.com/protocolbuffers/protobuf/blob/ae26a81918fa9e16f64ac27b5a2fb2b110b7aa1b/java/core/src/main/java/com/google/protobuf/Descriptors.java#L286
+   **/
+  private def accumulatedBuildFrom(
+      reads: Either[ReadFailure, Iterable[Descriptors.FileDescriptor]],
+      file: DescriptorProtos.FileDescriptorProto
+  ): Either[ReadFailure, Iterable[Descriptors.FileDescriptor]] = {
+    reads match {
+      case Left(_) => reads
+      case Right(fileDescriptors) =>
+        try {
+          Right(fileDescriptors ++ List(Descriptors.FileDescriptor.buildFrom(file, fileDescriptors.toArray, true)))
```

Comment:
So we have to pass 'depended-upon' descriptors when parsing 'depending' descriptors - meaning the files must be accumulated in the 'right order'. How do we make sure they are in the right order?

Reply:
I'm not sure they have to be in the right order, but maybe we should `allowUnknownDependencies=false` instead?

Reply:
> I'm not sure they have to be in the right order

Whatever descriptor is the first one will be called with an empty `fileDescriptors` array, so if some of them need things to be available in there then that'd mean the order matters?

> I'm not sure they have to be in the right order, but maybe we should `allowUnknownDependencies=false` instead?

I'm not sure, I guess you could have unknown dependencies (like external proto annotations that we don't care about) that we could allow?

Reply:
[This comment](https://github.com/protocolbuffers/protobuf/blob/ae26a81918fa9e16f64ac27b5a2fb2b110b7aa1b/java/core/src/main/java/com/google/protobuf/Descriptors.java#L301-L309) is what I expected, but the API doesn't seem to allow us to first accumulate the descriptors and after that trigger linking. 

I assume `allowUnknownDependencies=false` would make it fail when the order is wrong, but now the placeholders will be linked once the correct descriptor has been added to the list.